### ISSUE TITLE
Customize cillium cni configuration

### ISSUE THREAD

**Issue by vivek-koppuru** [aws/eks-anywhere#393]
**What would you like to be added**:
As mentioned in https://github.com/aws/eks-anywhere/issues/362, we want the ability to be able to modify configuration of the cni being installed (for example cilium in this case) before installing it on EKS-A. Currently, EKS-A deploys fixed version of cilium that can only be modified after the fact, which is currently not supported either.

**Why is this needed**:
Different users may have different use cases for configuration when deploying a cni on their EKS-A cluster. The option for debugging is a possibility as well for why a user might want to edit the cni configuration beforehand.

**Comment by cryptk** [comment]
It would be nice to be able to set the cluster-id and cluster-name parameters as well.  Ideally we would be able to specify any field in the cilium-config configmap, but at least being able to set those two would should go a long way towards using Cilium ClusterMesh

**Comment by mazzy89** [comment]
Perhaps buried in the documentation, but I'm wondering which are the reasons why EKS-A uses a non-vanilla Cilium Docker image? At the moment, the Cilium version is pretty old. At the time of writing, we are already at 1.12, and the EKS-A spins up cluster running Cilium 1.9.

**Comment by vincentmli** [comment]
@mazzy89 @vivek-koppuru I agree Cilium 1.9 is too old, that is why I suggested if not shipping most recent version, at least give users option to select cilium versions to deploy

**Comment by mazzy89** [comment]
@vincentmli the problem is if we had the option to select the cilium version, would be still ok? The Cilium version EKS-A runs is not the vanilla one, is it? EKS-A team bakes their own image. Not sure why though. I haven't looked at it.

**Comment by vincentmli** [comment]
> @vincentmli the problem is if we had the option to select the cilium version, would be still ok? The Cilium version EKS-A runs is not the vanilla one, is it? EKS-A team bakes their own image. Not sure why though. I haven't looked at it.

my goal is to test EKS-A with vanilla cilium version or most recent release version, I have not checked what cilium version  EKS-A runs, seems you were saying it runs with cilium 1.9 which is old.

**Comment by mazzy89** [comment]
@vincentmli did you run the test with a vanilla version?

**Comment by vincentmli** [comment]
@mazzy89 no, not yet, does EKS-A include cilium version 1.12 now?

**Comment by mazzy89** [comment]
Still old version

**Comment by vincentmli** [comment]
from what I heard from one user that similar product Google Anthos will include Cilium 1.12

**Comment by vincentmli** [comment]
I also wonder  if https://github.com/cilium/cilium/issues/21486 is the reason eks-a not shipping cilium 1.12, @vivek-koppuru do you have idea?

**Comment by vivek-koppuru** [comment]
Yea we don't maintain those images so we have to support what's available. @vincentmli

**Comment by vivek-koppuru** [comment]
Is this something that is blocking you?
